static/app/components/replaysOnboarding/useCurrentProjectState.tsx

```diff
@@ -1,99 +0,0 @@
-import {useEffect, useMemo, useState} from 'react';
-
-import {splitProjectsByReplaySupport} from 'sentry/components/replaysOnboarding/utils';
-import {SidebarPanelKey} from 'sentry/components/sidebar/types';
-import {replayOnboardingPlatforms, replayPlatforms} from 'sentry/data/platformCategories';
-import PageFiltersStore from 'sentry/stores/pageFiltersStore';
-import {useLegacyStore} from 'sentry/stores/useLegacyStore';
-import type {Project} from 'sentry/types';
-import useProjects from 'sentry/utils/useProjects';
-
-function useCurrentProjectState({currentPanel}: {currentPanel: '' | SidebarPanelKey}) {
-  const [currentProject, setCurrentProject] = useState<Project | undefined>(undefined);
-  const {projects, initiallyLoaded: projectsLoaded} = useProjects();
-  const {selection, isReady} = useLegacyStore(PageFiltersStore);
-
-  const isActive = currentPanel === SidebarPanelKey.REPLAYS_ONBOARDING;
-
-  // Projects where we have the onboarding instructions ready:
-  const projectsWithOnboarding = useMemo(
-    () =>
-      projects.filter(
-        p => p.platform && replayOnboardingPlatforms.includes(p.platform) && !p.hasReplays
-      ),
-    [projects]
-  );
-
-  // Projects that support replays, but we haven't created the onboarding instructions (yet):
-  const projectWithReplaySupport = useMemo(
-    () =>
-      projects.filter(
-        p => p.platform && replayPlatforms.includes(p.platform) && !p.hasReplays
-      ),
-    [projects]
-  );
-
-  useEffect(() => {
-    if (!isActive) {
-      setCurrentProject(undefined);
-    }
-  }, [isActive]);
-
-  useEffect(() => {
-    if (currentProject || !projectsLoaded || !projects.length || !isReady || !isActive) {
-      return;
-    }
-
-    if (!projectWithReplaySupport) {
-      return;
-    }
-
-    if (selection.projects.length) {
-      const selectedProjectIds = selection.projects.map(String);
-      // If we selected something that has onboarding instructions, pick that first
-      const projectForOnboarding = projectsWithOnboarding.find(p =>
-        selectedProjectIds.includes(p.id)
-      );
-      if (projectForOnboarding) {
-        setCurrentProject(projectForOnboarding);
-      }
```

Comment:
This was probably a bug before! The new code is returning if we get inside this if-statement, nice to see it fixed.